minorminer/utils/raster_embedding.py

```diff
@@ -0,0 +1,126 @@
+        **kwargs):
+    """
+    Searches for embeddings within a rastered subgraph.
+
+    Args:
+        _A (networkx.Graph): The hardware graph.
+        subgraph (networkx.Graph): The subgraph to embed.
+        gridsize (int, optional): Grid size. Defaults to 0.
+        raster_breadth (int, optional): Raster breadth. Defaults to 5.
+        verbose (bool, optional): Whether to print progress messages. Defaults to True.
+        topology (str, optional): The topology type ('chimera', 'pegasus', 'zephyr'). Defaults to 'pegasus'.
+        greed_depth (int, optional): Depth of greedy improvement. Defaults to 0.
+        verify_embeddings (bool, optional): Whether to verify embeddings. Defaults to True.
+        max_number_of_embeddings (int, optional): Maximum number of embeddings to find. Defaults to np.inf.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        np.ndarray: An embedding matrix.
+    """
+    _A = A.copy()
+
+    assert list(subgraph.nodes) == list(range(len(subgraph))), "Subgraph must have consecutive nonnegative integer nodes."
+
+    embs = []
+
+    if topology == 'chimera':
+        sublattice_mappings = dnx.chimera_sublattice_mappings
+        tile = dnx.chimera_graph(raster_breadth)
+    elif topology == 'pegasus':
+        sublattice_mappings = dnx.pegasus_sublattice_mappings
+        tile = dnx.pegasus_graph(raster_breadth)
+    elif topology == 'zephyr':
+        sublattice_mappings = dnx.zephyr_sublattice_mappings
+        tile = dnx.zephyr_graph(raster_breadth)
+    else:
+        raise ValueError(f"Unsupported topology: {topology}")
+    
+    for i, f in enumerate(sublattice_mappings(tile, _A)):
+        B = _A.subgraph([f(_) for _ in tile]).copy()
+
+        sub_embs = search_for_subgraphs_in_subgrid(B, subgraph,
+                                                   max_number_of_embeddings=max_number_of_embeddings,
+                                                   **kwargs)
+        
+        for sub_emb in sub_embs:
+            _A.remove_nodes_from(sub_emb.values())
+
+        if verify_embeddings:
+            for emb in sub_embs:
+                X = list(embedding.diagnose_embedding(
+                    {p: [emb[p]] for p in sorted(emb.keys())}, subgraph, A
+                ))
+                if X:
+                    raise Exception("Embedding verification failed.")
+
+        embs += sub_embs
+        if len(embs) >= max_number_of_embeddings:
+            break
+
+    # Get independent set of embeddings (removed, potential feature expansion)
+
+    embmat = np.asarray([[ie[v] for ie in embs] for v in sorted(subgraph.nodes)]).T
+
+    if verify_embeddings:
+        for emb in embmat:
+            X = list(embedding.diagnose_embedding({p: [emb[p]] for p in range(len(emb))}, subgraph, A))
+            if X:
+                raise Exception("Embedding verification failed.")
+
+    assert len(np.unique(embmat)) == len(embmat.ravel()), "Embeddings are not unique."
```

Comment:
We will move this to tests/  (not required to approve pull request)

Reply:
Should we still leave a try/catch block in the code?

Reply:
oh good point, test or try/except?

Reply:
this shouldn't happen, so probably better a test.